server/src/main/java/org/elasticsearch/node/Node.java

```diff
@@ -1079,18 +1080,24 @@ protected Node(
                         );
                 }
                 b.bind(HttpServerTransport.class).toInstance(httpServerTransport);
-                pluginComponents.forEach(p -> {
+                // TODO this is horrible business
```

Comment:
As discussed this is obviously bad hackery. For the final implementation, we will come up with a cleaner approach.